### ISSUE TITLE
teal framework Release preparation

### ISSUE THREAD

**Issue by donyunardi** [insightsengineering/nestdevs-tasks#66]
### Summary

We have completed the following enhancements
* #58 
* https://github.com/insightsengineering/teal/issues/669

Let's plan for a release of teal framework packages. We will handle teal modules packages release after this.

### Release candidate:
* https://github.com/insightsengineering/teal.logger/issues/94 
  * teal.logger v0.3.0 (log_shiny_input_changes)
* teal.reporter v0.4.0 (Archiver Module)
* teal.code v0.6.0 (introduce `names` to qenv)
* teal.widgets v0.5.1 (several bug fixes)
* teal.data v0.7.0 (deprecate datanames)
* teal.slice v0.6.0 (insertUI)
* teal.transform v0.5.1 (minor code update)
* teal v0.16.0 (teal_transform_module, insertUI, decorator)

#### teal.data
* https://github.com/insightsengineering/teal.data/issues/333
* https://github.com/insightsengineering/teal.code/issues/210
* https://github.com/insightsengineering/teal.code/issues/211
* https://github.com/insightsengineering/teal.data/issues/338

#### teal/teal.slice
* https://github.com/insightsengineering/teal/issues/1391
* https://github.com/insightsengineering/teal/issues/1321
* https://github.com/insightsengineering/teal/issues/1352
* https://github.com/insightsengineering/teal/issues/1347
* https://github.com/insightsengineering/teal/issues/1380
* https://github.com/insightsengineering/teal/issues/1378
* https://github.com/insightsengineering/teal/issues/1379
* https://github.com/insightsengineering/teal/issues/1366
* https://github.com/insightsengineering/teal/issues/1323
* https://github.com/insightsengineering/teal/issues/1271
* https://github.com/insightsengineering/teal/issues/1310
* https://github.com/insightsengineering/teal/issues/1312
* https://github.com/insightsengineering/teal/issues/1268
* https://github.com/insightsengineering/nestdevs-tasks/issues/86
* https://github.com/insightsengineering/teal/issues/1417

#### Deprecation activity
* https://github.com/insightsengineering/teal/issues/1386
* https://github.com/insightsengineering/teal.code/issues/215
* https://github.com/insightsengineering/teal.data/issues/344
* https://github.com/insightsengineering/teal.slice/issues/623

#### Deprecation activity
Make sure all CI pipelines are green before release (i.e. Scheduled pipeline, etc).

#### Others
* https://github.com/insightsengineering/coredev-tasks/issues/605

**Comment by m7pr** [comment]
I think we should do some manual testing on `teal.gallery` or/and `tlg-catalog` just in case.
Also review if all pipelines/workflows are green once we release (or before we release).

**Comment by m7pr** [comment]
@donyunardi should we revisit packages for soft deprecated functions and hard deprecate them? Also, for the hard deprecated functions, should we delete them? For example, should we delete all those teal.data functions that got hard deprecated 9 months ago ? https://github.com/insightsengineering/teal.data/blob/main/R/deprecated.R

**Comment by donyunardi** [comment]
Yeah, good idea. Let's review the deprecated functions in all packages.

> For example, should we delete all those teal.data functions that got hard deprecated 9 months ago

I would say yes, 9 months ago is a reasonable time for us to remove this.

**Comment by m7pr** [comment]
@donyunardi @kumamiao what do you think about planning a community meeting with the summary of what got changed during the release? We could record and put that on YouTube and also include the link to the video in the release notes?

**Comment by donyunardi** [comment]
Sure, I'm open for it. 👍🏼

**Comment by donyunardi** [comment]
> I think we should do some manual testing on teal.gallery or/and tlg-catalog just in case.

I believe our shinytest2 implementation on teal frameworks and modules covers most of the key areas, but I'm open to doing minimal tests on teal.gallery. For tlg-catalog, I need a better understanding of what requires manual testing.

@m7pr, since you were involved in the shinytest2 implementation, could you help outline any gaps where manual testing is needed for both teal.gallery and tlg-catalog before the release? 

> Also review if all pipelines/workflows are green once we release (or before we release)

Yes, let's make sure this happens.

**Comment by kumamiao** [comment]
> @donyunardi @kumamiao what do you think about planning a community meeting with the summary of what got changed during the release? We could record and put that on YouTube and also include the link to the video in the release notes?

Sure sounds good, was thinking the same as it's been a while since the last one. I'll be out Nov - Jan, so I was thinking sometime in Jan may be a good timing (assuming this will be aligning with the release timeline - right after the release?), but let's also see how the release work goes.

**Comment by donyunardi** [comment]
We'll continue with #96 
All open issues will be carried over to the new issue.
